src/fragments/lib/datastore/ios/data-access/update-snippet.mdx

```diff
@@ -1,18 +1,18 @@
 <BlockSwitcher>
 
-<Block name="Listener (iOS 11+)">
+<Block name="Async/Await">
 
 ```swift
 let existingPost: Post = /* get an existing post */
```

Comment:
Umm, I think in this context it can be `let` as we're not changing it anywhere later in the code snippet right?

Reply:
i believe the existing example is poorly communicated, since it assumes there's a post retrieved as a variable called `existingPost`. Even if we were to add how the existingPost is retrieved, the `let` will cause a compile error
```swift
guard let queriedPost = try await Amplify.DataStore.query(Post.self, byId: "postId") else {
            return
        }
        queriedPost.title = "updated"
```
the error `"Cannot assign to property: 'queriedPost' is a 'let' constant".`

So we should at least change it to `var`. Perhaps this is the time to replace `/* get an existing post */` as well since it's no longer a callback method taking up multiple lines and we can easily write it out as:
```swift
guard var queriedPost = try await Amplify.DataStore.query(Post.self, byId: "postId") else {
            return
        }
        queriedPost.title = "updated"
```